### PR TITLE
tokio: rewrite io_read.rs test to use async/await

### PR DESCRIPTION
**Pull request by carllerche**
This simplifies the test.

### CODE REVIEW

**Comment by carllerche** [comment]
I went ahead and did it.

File: tokio/tests/io_read.rs [hunk 1]
```diff
@@ -1,41 +1,38 @@
 #![deny(warnings, rust_2018_idioms)]
+#![feature(async_await)]
```
Comment (LucioFranco) [inline tokio/tests/io_read.rs]: we could prob just throw this in a mod under the read.rs?

Comment (carllerche) [reply tokio/tests/io_read.rs]: I think it has to be in each test file.

Comment (LucioFranco) [reply tokio/tests/io_read.rs]: sorry I mean, the whole test.

**Pull request merged**